bluesky_widgets/models/plot_builders.py

```diff
@@ -233,6 +245,49 @@ def restyle_line_when_complete(event):
             line = Line.from_run(func, run, label, style)
             self._run_manager.track_artist(line, [run])
             self.axes.artists.append(line)
+            self._ys_to_artists[y].append(line)
+
+    def _add_ys(self, event):
+        # Update title and y_label when adding a new y
+        self.y_label = ", ".join(auto_label(y) for y in self.ys)
+        self.title = f"{self.axes.y_label} v {self.axes.x_label}"
+        y = event.item
+        if y in self._ys_to_artists.keys():
+            # Don't plot a new line if y already exists
```

Comment:
I'm worried this could create unexpected behavior if someone _wants_ to plot the same thing twice or some other unforeseen scenario. It just feels odd to have the let `self.ys` get out of sync with `self._ys_to_artists`. If we think it's important to protect users from plotting something twice, I'd suggest we should block them from doing so further up---i.e. reject the addition to `self.ys` in the first place by giving it the semantics of an `OrderedSet` rather than a list. (That might be something to consider in a separate issue/PR.)